shim/src/hashtable.c

```diff
@@ -120,7 +117,7 @@ void hashtable_remove(struct hashtable *h, int key)
             return;
         }
 
-        hash = (hash + 1) & mask;
+        hash = hash + 1;
```

Comment:
We should keep modulo operation here, otherwise we may end up writing outside the bounds of the array.